scripts/genproto.sh

```diff
@@ -25,8 +25,25 @@ fi
 source ./scripts/test_lib.sh
 
 if [[ $(protoc --version | cut -f2 -d' ') != "3.20.3" ]]; then
-  echo "could not find protoc 3.20.3, is it installed + in PATH?"
-  exit 255
+  echo "Could not find protoc 3.20.3, installing now..."
+
+  arch=$(go env GOARCH)
+
+  case ${arch} in
+    "amd64") file="x86_64" ;;
+    "arm64") file="aarch_64" ;;
+    *)
+      echo "Unsupported architecture: ${arch}"
+      exit 255
+      ;;
+  esac
+
+  download_url="https://github.com/protocolbuffers/protobuf/releases/download/v3.20.3/protoc-3.20.3-linux-${file}.zip"
+  echo "Running on ${arch}."
+  wget ${download_url} && unzip protoc-3.20.3-linux-${file}.zip
+  mv bin/protoc /usr/local/bin/ && chmod +x /usr/local/bin/protoc
```

Comment:
Just want to note that this will fail without sudo. 

I think it would be less invasive to just `chmod` the file in the current working directory.

Reply:
Agree, we shouldn't require sudo.

Reply:
Updated thanks